internal/webhooks/cluster.go

```diff
@@ -305,6 +305,7 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 	// If there's no error validate the Cluster based on the ClusterClass.
 	if clusterClassPollErr == nil {
 		allErrs = append(allErrs, ValidateClusterForClusterClass(newCluster, clusterClass)...)
+		allErrs = append(allErrs, validateAutoscalerAnnotationsForCluster(newCluster, clusterClass)...)
```

Comment:
Sorry that was probably misleading. This should go one line down (directly after the if)

Because we also want to run it if we couldn't get the ClusterClass

Reply:
updated. https://github.com/kubernetes-sigs/cluster-api/compare/1edc35daea5eb2c9718ba69495a897f84ef73fb5..e035135b77573d5760bbf647a20b4c3c37e4c1c8